web/src/beta/components/Resizable/index.tsx

```diff
@@ -21,15 +24,27 @@ const Resizable: React.FC<Props> = ({
   maxSize,
   initialSize,
   children,
+  localStorageKey,
+  tab,
 }) => {
-  const { size, gutterProps, minimized, handleResetSize } = useHooks(
+  const { size, gutterProps, minimized, handleResetSize, setSizeFromStorage } = useHooks(
     direction,
     gutter,
     initialSize,
     minSize,
     maxSize,
+    localStorageKey,
   );
 
+  useEffect(() => {
+    const setSizeBasedOnTab = () => {
+      if (tab === "map" || tab === "story" || tab === "widgets") {
+        localStorageKey && setSizeFromStorage(localStorageKey);
+      }
+    };
+    setSizeBasedOnTab();
+  }, [tab, setSizeFromStorage, localStorageKey]);
```

Comment:
First, do we really need to check the tab? Or can we just `if (!!localStorageKey) setSizeFromStorage(localStorageKey)`?
Second, Resizable is a reusable component, so shouldn't have such specific logic here anyways.

Instead of using a useEffect and the usecallback (setSizeFromStorage), you could probably just use a useMemo, no? 🤔